core/src/style/polygonStyle.cpp

```diff
@@ -58,16 +77,14 @@ void PolygonStyle::constructShaderProgram() {
     std::string fragShaderSrcStr = stringFromFile("shaders/polygon.fs", PathType::internal);
 
     m_shaderProgram->setSourceStrings(fragShaderSrcStr, vertShaderSrcStr);
+
+    m_shaderProgram->addSourceBlock("defines", m_defines);
```

Comment:
Instead of maintaining member for `defines`, we can check for `m_texCoordsGeneration` and explicitly add the define block: `#define TANGRAM_USE_TEX_COORDS\n`